storage/state.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2019-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package storage
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/ava-labs/avalanchego/utils/perms"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	stateFile = "apm.state"
+)
+
+func NewEmptyStateFile(path string) StateFile {
+	return StateFile{
+		Sources:      make(map[string]SourceInfo),
+		RepoList:     make(map[string]RepoList),
+		InstalledVMs: make(map[string]InstallInfo),
+		path:         filepath.Join(path, stateFile),
+	}
+}
+
+func NewStateFile(path string) (StateFile, error) {
+	result := NewEmptyStateFile(path)
+
+	b, err := os.ReadFile(result.path)
+	var pathError *os.PathError
+	if errors.As(err, &pathError) {
+		// need to initialize the StateFile on Commit
```

Comment:
do you need to do something here?

Reply:
Yeah this needs to return an err, forgot to mark a TODO. I'll update this before I merge